### ISSUE TITLE
Null ref: T2.StaffModule.ResetPositionAndRotation

### ISSUE THREAD

**Issue by ChrisNZL** [ChrisNZL/Tallowmere2#969]
Auto report. v0.2.9. Feedback ID: 20211024-B6N5K

```
4:03:40, Frame 275469, LOG »  DungeonRoomTransitioner.BeginTransitioningToRoom: Departing room 0...
4:03:42, Frame 275517, LOG »  DungeonRoomTransitioner.Update: Revealing room: 第37个房间：蓝区


4:04:42, Frame 279090, EXCEPTION »  NullReferenceException: Object reference not set to an instance of an object
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

T2.StaffModule.ResetPositionAndRotation ()
T2.ElementalStaff.ResetToDefaultState ()
T2.ElementalStaff.OnParentCreatureBecameStunned ()
T2.Debuff_Stunned._OnAppliedToCreatureStruct (T2.CreatureStruct creatureStruct)
T2.Debuff.OnAppliedToCreatureStruct (T2.CreatureStruct creatureStruct)
T2.CreatureStruct.__ApplyDebuff (T2.PendingDebuffData debuffData)
T2.CreatureBody.AddPendingDebuffs ()
T2.CreatureBody.Update ()

GameStates: RompingThroughDungeon
OnlineGameInfo: Device is offline.
No previous server nor online game info.
GameSetupMode: Singleplayer
Players: 1
DungeonRoomTransitionerState: Null
Room: 37 / 37
RoomModifiers: ExtraKnockback (击退效果加强), ArcaneNightmare (奥术梦魇), Heartless (无心无情), Penniless (家徒四壁), RangedOnly (远程练习)
YouHaveDiedState: Null
SystemPlayer InputDevice: 键盘
HumanPlayer 1 InputDevice: 键盘
```

**Comment by ChrisNZL** [comment]
Further auto report, 0.2.10a, feedback ID 20211108-85N3S

Error while loading room

```
4:07:41, Frame 498736, LOG »  DungeonRoomTransitioner.BeginTransitioningToRoom: Departing room 116, going to room 117...

4:07:43, Frame 498810, LOG »  NetworkMessage_DungeonRoomTransitionHelper: Received message.
Desire: 4 » BeginLoadingRoom

4:07:43, Frame 498823, EXCEPTION »  NullReferenceException: Object reference not set to an instance of an object
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

T2.StaffModule.ResetPositionAndRotation ()
T2.ElementalStaff.ResetToDefaultState ()
T2.Weapon._OnBecameEquipped ()
T2.InventoryItem.OnBecameEquipped ()
T2.Creature.HoldInstantiatedWeapon (T2.Weapon weapon)
T2.Creature.InstantiateAndHoldInventoryItemFromItemStruct (T2.InventoryItemStruct itemStruct)
T2.Creature._EquipWeaponsAndItems (T2.CreatureStruct creatureStruct, T2.MethodCaller methodCaller)
T2.Creature.InstantiateCreature (T2.CreatureStruct creatureStruct, T2.SeedFactory seedFactory, System.Boolean allowAI)
T2.DungeonRoom.Update ()

GameStates: RompingThroughDungeon, TransitioningToDungeonRoom
OnlineGameInfo: Device is online.
Server: Lightning-Beijing-01
GameID: CFR4F
GameSetupMode: NetworkCoop
Players: 2
DungeonRoomTransitionerState: WaitingForRoomToLoad
Room: 117 / 116
RoomModifiers: FromTheGrave (破坟而起), HugeProjectiles (巨型抛射物), Penniless (家徒四壁), Bouncy (弹力十足), StunBombsOnly (晕眩炸弹雨林)
YouHaveDiedState: Null
SystemPlayer InputDevice: XInput Controller
HumanPlayer 1 InputDevice: XInput Controller
```

**Comment by ChrisNZL** [comment]
Another auto report, v0.2.11, feedback ID 20211129-Q5M57

```
5:21:23, Frame 234472, LOG »  DungeonRoomTransitioner.Update: Revealing room: 第635个房间：烈火洞窟

5:21:32, Frame 234769, EXCEPTION »  NullReferenceException: Object reference not set to an instance of an object
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

T2.StaffModule.ResetPositionAndRotation ()
T2.ElementalStaff.ResetToDefaultState ()
T2.ElementalStaff.OnParentCreatureBecameStunned ()
T2.Debuff_Stunned._OnAppliedToCreatureStruct (T2.CreatureStruct creatureStruct)
T2.Debuff.OnAppliedToCreatureStruct (T2.CreatureStruct creatureStruct)
T2.CreatureStruct.__ApplyDebuff (T2.PendingDebuffData debuffData)
T2.CreatureBody.AddPendingDebuffs ()
T2.CreatureBody.Update ()

GameStates: RompingThroughDungeon
OnlineGameInfo: Device is offline.
No previous server nor online game info.
GameSetupMode: Singleplayer
Players: 1
DungeonRoomTransitionerState: Null
Room: 635 / 635
RoomModifiers: DoubleTrouble (双倍麻烦), ElementalStaffsOnly (咒术派对), Penniless (家徒四壁), NoWaypoint (没有传送点), ExtraChests (更多宝箱)
YouHaveDiedState: Null
SystemPlayer InputDevice: 键盘
HumanPlayer 1 InputDevice: 键盘
```

**Comment by ChrisNZL** [comment]
Fixed in 0.3.0. Added null checks.

**Comment by ChrisNZL** [comment]
Another auto report. v0.3.0.

One last null check added now for v0.3.0a.

If this still errors, will have to wrap the lot in a try/catch.

```csharp
public void ResetPositionAndRotation () {
	if (transform.Exists()) {
		if (weapon.Exists()) {
			Creature creature = weapon.creature;
			if (creature.Exists()) {
				CreatureBody creatureBody = creature.body;
				if (creatureBody.Exists()) {
					Transform staff_idlePoint = creatureBody.staff_idlePoint;
					if (staff_idlePoint.Exists()) {
						transform.position = staff_idlePoint.position;
					}
				}
			}
		}
		transform.localRotation = db2.staffRotation_idle;
	}
	nearestCreature = null;
}
```
